### ISSUE TITLE
fundamental.cc can't tracking ceres when building

### ISSUE THREAD

**Issue by jack7141** [opencv/opencv_contrib#2030]
- OpenCV => 3.4.0
-OpencCV_Contrib =>3.4.0
- Operating System / Platform => window 10 64bit
- Compiler => Mingw 32bit

##### Detailed description
when i building can't tracking ceres/ceres.h
i already installed ceres,eigen,glog,gflag ..
why occured this happend??

[ 94%] Building CXX object modules/sfm/src/libmv/libmv/multiview/CMakeFiles/multiview.dir/fundamental.cc.obj
C:\dev\opencv_contrib-3.4.4\modules\sfm\src\libmv_light\libmv\multiview\fundamental.cc:24:25: fatal error: ceres/ceres.h: No such file or directory
compilation terminated.
modules\sfm\src\libmv\libmv\multiview\CMakeFiles\multiview.dir\build.make:90: recipe for target 'modules/sfm/src/libmv/libmv/multiview/CMakeFiles/multiview.dir/fundamental.cc.obj' failed
mingw32-make[2]: *** [modules/sfm/src/libmv/libmv/multiview/CMakeFiles/multiview.dir/fundamental.cc.obj] Error 1
CMakeFiles\Makefile2:5337: recipe for target 'modules/sfm/src/libmv/libmv/multiview/CMakeFiles/multiview.dir/all' failed
mingw32-make[1]: *** [modules/sfm/src/libmv/libmv/multiview/CMakeFiles/multiview.dir/all] Error 2
Makefile:161: recipe for target 'all' failed
mingw32-make: *** [all] Error 2

**Comment by PNeigel** [comment]
I have the same problem. Ceres was installed correctly I guess (tests were all successful), ceres.h is there in the ceres include directory. Why is it not included when building the multiview lib of sfm?

I build Ceres according to https://docs.opencv.org/4.0.1/db/db8/tutorial_sfm_installation.html, specifially as a *static* library (by default). Does that have anything to do with the problem?

Building OpenCV without the SFM module works perfectly fine.

**Comment by PNeigel** [comment]
I found a workaround:
In the opencv_contrib sources find the file opencv_contrib/modules/sfm/SMakeLists.txt. Line 60 says:
    list(APPEND LIBMV_LIGHT_INCLUDES "${CERES_INCLUDE_DIR}")

**I think this behaviour is a bug and $(CERES_INCLUDE_DIR) is not defined for whatever reason**(cmake found the ceres installation without problems, so it should be there)
I added the following line directly after:
    list(APPEND LIBMV_LIGHT_INCLUDES /path/to/your/ceres-install/include)

Now it build and installed without problems.

**Comment by jack7141** [comment]
> I found a workaround:
> In the opencv_contrib sources find the file opencv_contrib/modules/sfm/CMakeLists.txt.
> Line 60 says:
> `list(APPEND LIBMV_LIGHT_INCLUDES "${CERES_INCLUDE_DIR}")`
> 
> **I think this behaviour is a bug and $(CERES_INCLUDE_DIR) is not defined for whatever reason**(cmake found the ceres installation without problems, so it should be there)
> I added the following line directly after:
> `list(APPEND LIBMV_LIGHT_INCLUDES /path/to/your/ceres-install/include)`
> where /path/to/your/ceres-install is of course the path to your ceres install
> 
> Now it built and installed without problems.

**Comment by jack7141** [comment]
Thank you!! have a good day!!

**Comment by PNeigel** [comment]
Just to add to this: Today I was trying to build openCV 4.1.0 with the SFM Module. I configured with CMake 3.14.1 and when building in Visual Studio 2017 the `multiview` module fails to build for the very same reason. I had to add the Ceres include directory manually to the multiview module so it could build successfully, so I think this is definitely some problem with the CMake files. This issue should be **open**, not closed.
